src/sql/db_connection_pool/dbconnection/sqliteconn.rs

```diff
@@ -63,7 +63,7 @@ impl AsyncDbConnection<Connection, &'static (dyn ToSql + Sync)> for SqliteConnec
                 let mut stmt = conn.prepare(&format!("SELECT * FROM {table_reference} LIMIT 1"))?;
                 let column_count = stmt.column_count();
                 let rows = stmt.query([])?;
-                let rec = rows_to_arrow(rows, column_count)
+                let rec = rows_to_arrow(rows, column_count, None)
```

Comment:
For acceleration (write) mode this is not used so there is no such issue. For SQLite data connector mode this should be improved to rely on native schema inference instead of using first row